### PR TITLE
[FEATURE] Collect code coverage and add a coverage badge

### LINKED ISSUE #807: Add coveralls.io integration
- [x] ~add coveralls to repo~ #808 
- [x] enable coveralls on organisation
- [x] add badge to readme

re: https://github.com/MyIntervals/emogrifier/pull/804#issuecomment-539230157

to avoid getting feature PRs that don't include coverage.

### LINKED ISSUE #1126: Composer script for PHPUnit coverage report
I've read [this](https://phpunit.readthedocs.io/en/9.5/code-coverage-analysis.html), and still don't know how to get the report.  I'm sure it's a simple command line option, but can't find easily find it.

TBH, I was barely aware this PHPUnit feature existed.

Unless I'm mistaken, we should add a Composer script (command) to run PHPUnit with whatever command line option is needed to generate the report.  This will prompt developers on the project to have a look at it, perhaps initially out of idle curiosity, but potentially leading to improvements in the robustness of the unit tests.

### PR DESCRIPTION
**Pull request by oliverklee**
Fixes #1126
Fixes #807

### CODE REVIEW

**Comment by oliverklee** [comment]
I'll still need to change the path to the coverage file: https://github.com/php-coveralls/php-coveralls/issues/251

**Comment by oliverklee** [comment]
Note to myself: https://github.com/CPS-IT/project-builder/pull/500/files

**Comment by oliverklee** [comment]
Note: As far as I understand it, Coveralls requires the coverage job to be run for the `main` branch at least once before it will start to post the change to the coverage numbers on PRs.

**Comment by oliverklee** [comment]
We might actually be able to reuse things after all. I've created #1286 for this. (I'd like to tackle this later once this PR here is merged, though.)

File: .gitignore [hunk 1]
```diff
@@ -23,6 +23,7 @@ nbproject
 /.php_cs.cache
 /.php-cs-fixer.cache
 /.phpunit.result.cache
+/coverage.xml
```
Comment (JakeQZ) [inline .gitignore:26]: I'd like some of these type of files to be squirrelled away in a subdirectory, though that is beyond the scope of this PR.

**Review by JakeQZ — APPROVED** [review]
Looks OK.  If there are any issues, we can resolve them later.

**Pull request merged**